crates/pathfinder/src/state/sync.rs

```diff
@@ -913,6 +918,14 @@ async fn l2_update(
             .insert_state_update(block.block_number, &state_update)
             .context("Insert state update into database")?;
 
+        transaction
+            .update_state_diff_commitment_and_length(
+                header.number,
+                state_diff_commitment,
+                state_update.state_diff_length(),
+            )
+            .context("Inserting state diff length into database")?;
+
```

Comment:
Can't we simply make this part of `insert_block_header()`?

Reply:
Lemme raise an issue for refactoring the `BlockHeader/SignedBlockHeader` duo.